### PR TITLE
task3

### PR DESCRIPTION
**Pull request by alexander-hizels**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -1,43 +1,52 @@
...
+    <script src="app.js"></script>
+    <script src="src/pages/home-page/home.js"></script>
+    <script src="src/components/course/courseDirective.js"></script>
+    <script src="src/components/course/formatDurationFilter.js"></script>
+    <script src="src/components/border-date/borderDateDirective.js"></script>
+    <script src="src/components/course-form/courseFormDirective.js"></script>
+    <script src="src/services/coursesService.js"></script>
+    <script src="src/services/searchDataTransferService.js"></script>
+    <script src="src/components/courses/coursesDirective.js"></script>
+    <script src="src/components/courses/orderByPropFilter.js"></script>
```
Comment (kirillparamonov) [inline app/index.html]: Please check this import. As I can see, application can't find this file:
image

File: app/src/components/border-date/borderDateDirective.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+var COLORS = { blue: 'blue', green: 'green' };
+var MLSEC_A_DAY = 86400000;
+
+angular.module('appBorderDateDirective', [])
+    .directive('appBorderColor', function() {
+        return {
+            scope: false,
+            link: function(scope, element, attrs) {
```
Comment (kirillparamonov) [inline app/src/components/border-date/borderDateDirective.js]: Please remove unused parameter attrs.

File: app/src/components/border-date/borderDateDirective.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+                var date = scope.course.date.getTime();
+                var now = Date.now();
```
Comment (kirillparamonov) [inline app/src/components/border-date/borderDateDirective.js]: I suggest to rename both variables to courseDate(or courseCreationDate) and currentDate

File: app/src/components/courses/courses.tpl.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<section ng-controller="CoursesController" class="courses courses__container">
+    <app-course ng-repeat="course in getCourses() | limitTo:coursesLimit
+                                                  | filter: getFilterValue()
+                                                  | orderBy: orderByDate"
+                data="course"
+                ng-if="getCourses()"></app-course>
```
Comment (kirillparamonov) [inline app/src/components/courses/courses.tpl.html]: As I see, you have three ng-if and one ng-repeat that uses getCourses() here. I believe that these calls can be moved to controller variable. $scope.courses = getCourses()

Comment (alexander-hizels) [reply app/src/components/courses/courses.tpl.html]: OK. I fixed it.

File: app/src/components/border-date/borderDateDirective.js (hunk 3 above)
Comment (alexander-hizels) [reply app/src/components/border-date/borderDateDirective.js]: OK. I fixed it.

File: app/src/components/border-date/borderDateDirective.js (hunk 2 above)
Comment (alexander-hizels) [reply app/src/components/border-date/borderDateDirective.js]: OK. I fixed it.

File: app/index.html (hunk 1 above)
Comment (alexander-hizels) [reply app/index.html]: OK. I fixed it.

**Pull request merged**